indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -214,309 +214,307 @@ public void unregisterListener(String listenerId)
   public ListenableFuture<TaskStatus> run(final Task task)
   {
     synchronized (tasks) {
-      if (!tasks.containsKey(task.getId())) {
-        tasks.put(
-            task.getId(),
-            new ForkingTaskRunnerWorkItem(
-                task,
-                exec.submit(
-                    new Callable<TaskStatus>()
-                    {
-                      @Override
-                      public TaskStatus call()
-                      {
-                        final String attemptUUID = UUID.randomUUID().toString();
-                        final File taskDir = taskConfig.getTaskDir(task.getId());
-                        final File attemptDir = new File(taskDir, attemptUUID);
-
-                        final ProcessHolder processHolder;
-                        final String childHost = node.getHost();
-                        int childPort = -1;
-                        int tlsChildPort = -1;
-
-                        if (node.isEnablePlaintextPort()) {
-                          childPort = portFinder.findUnusedPort();
+      tasks.putIfAbsent(
+          task.getId(),
+          new ForkingTaskRunnerWorkItem(
+            task,
+            exec.submit(
```

Comment:
`exec.submit` is a side effect, so this should not use `putIfAbsent`.

Reply:
Done.